test/SILOptimizer/generalized_accessors.swift

```diff
@@ -273,25 +299,77 @@ struct TestYieldsInGuards {
 }
 
 struct TestYieldsInGuards2 {
+  var flag: Bool
+  var defaultStorage: Int
+
+  var computed: Int {
+    _read {
+      guard flag else { // expected-note {{must yield when the condition is false}}
+        return
+      }
+      yield defaultStorage
+    } // expected-error {{accessor must yield on all paths before returning}}
+
+    _modify {
+      guard flag else { // expected-note {{must yield when the condition is true}}
+        yield &defaultStorage
+        return
+      }
+      defaultStorage += 1
+    } // expected-error {{accessor must yield on all paths before returning}}
+  }
+}
+
+struct TestYieldsInLetPatterns {
   var storedOpt: Int?
   var defaultStorage: Int
 
   var computed: Int {
     _read {
-      guard let stored = storedOpt else {
+      if let stored = storedOpt { // expected-note {{must yield in the case 'none'}}
```

Comment:
Can we special case this diagnostic text for Optional so that we say "in the nil case" and in the "in the non-nil case"?

I think most Swift programmers don't think in terms of .none and .some() but rather as nil and non-nil.